metrics/codahale.go

```diff
@@ -100,10 +103,24 @@ func (c *CodaHale) updateTimer(key string, d time.Duration) {
 	}
 }
 
+func (c *CodaHale) getHistogram(key string) metrics.Histogram {
+	return c.reg.GetOrRegister(key, c.createHistogram).(metrics.Histogram)
+}
+
+func (c *CodaHale) updateHistogram(key string, value float64) {
+	if t := c.getHistogram(key); t != nil {
+		t.Update(int64(value))
+	}
+}
+
 func (c *CodaHale) MeasureSince(key string, start time.Time) {
 	c.measureSince(key, start)
 }
 
+func (c *CodaHale) MeasureFloat(key string, value float64) {
+	go c.updateHistogram(key, value)
```

Comment:
why do you spawn a goroutine?

Reply:
that was the same as in measure since

Reply:
Where do we call MeasureFloat ?
If this is not in the hot path of the proxy package, then I would say drop the goroutine.